skfp/descriptors/topological.py

```diff
@@ -45,6 +46,102 @@ def average_wiener_index(
     return (2 * wiener_idx) / (num_atoms * (num_atoms - 1))
 
 
+def balaban_j_index(mol: Mol, distance_matrix: Optional[np.ndarray] = None) -> float:
+    """
+    Balaban's J Index.
+
+    Calculates the Balaban’s J index [1]_ [2]_, defined as a measure of molecular
+    connectivity with an emphasis on cyclic structures. The formula for calculating
+    Balaban's J Index is given as:
+
+    .. math::
+
+        J = \\frac{M}{μ + 1} \\cdot \\frac{Σ_{ij} (d_{ij})^{-1}}{n},
+
+    where:
+
+    - :math:`M` is the number of edges in the molecular graph,
+    - :math:`μ` is the cyclomatic number (number of independent cycles),
+    - :math:`d_{ij}` is the distance between atoms :math:`i` and :math:`j`,
+    - :math:`n` is the number of atoms in the molecule.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the Balaban's J index is to be calculated.
+
+    distance_matrix : np.ndarray, optional
+        Precomputed distance matrix. If not provided, it will be calculated by RDKit.
+
+    References
+    ----------
+    .. [1] `Balaban, Alexandru T.
+        "Highly discriminating distance-based topological index."
+        Chemical physics letters 89.5 (1982): 399-404.
+        <https://www.sciencedirect.com/science/article/abs/pii/0009261482800092>`_
+
+    .. [2] `<https://www.rdkit.org/docs/source/rdkit.Chem.GraphDescriptors.html>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.topological import balaban_j_index
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> balaban_j_index(mol)
+    3.000000000000001
+    """
+    return BalabanJ(mol=mol, dMat=distance_matrix)
+
+
+def diameter(mol: Mol, distance_matrix: Optional[np.ndarray] = None) -> int:
+    """
+    Diameter.
+
+    Calculates the diameter [1]_, defined as the maximum eccentricity among all atoms
+    in the molecular graph. The eccentricity of an atom is the greatest shortest
+    path distance to any other atom. It is given as:
+
+    .. math::
+
+        D = \\max_i(η_i)
+
+    where:
+
+    - :math:`η_i = \\max_j(d_{ij})` is the eccentricity of the :math:`i^{th}` atom.
+    - :math:`d_{ij}` is the shortest path distance between atoms :math:`i` and :math:`j`.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the diameter is to be calculated.
+
+    distance_matrix : np.ndarray, optional
+        Precomputed distance matrix. If not provided, it will be calculated.
+
+    References
+    ----------
+    .. [1] `Petitjean, Michel.
```

Comment:
Why dot after author name? Why lowercase name for journal name? Why such a long link instead of short DOI?